internal/cmd/fetcher/main.go

```diff
@@ -119,6 +122,97 @@ func recreateNPMPackageLock(ctx context.Context, plugin createdPlugin) error {
 	return cmd.Run()
 }
 
+// recreateSwiftPackageResolved resolves Swift package dependencies for plugins that use Swift packages.
+// It clones the git repository specified in the Dockerfile, runs 'swift package resolve',
+// and moves the generated Package.resolved file to the version directory.
+func recreateSwiftPackageResolved(ctx context.Context, plugin createdPlugin) error {
+	versionDir := filepath.Join(plugin.pluginDir, plugin.newVersion)
+	packageResolved := filepath.Join(versionDir, "Package.resolved")
+	_, err := os.Stat(packageResolved)
+	if err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			return err
+		}
+		// no Package.resolved to update
+		return nil
+	}
+
+	// Read the Dockerfile to find the git clone command
+	dockerfile := filepath.Join(versionDir, "Dockerfile")
+	file, err := os.Open(dockerfile)
+	if err != nil {
+		return fmt.Errorf("failed to open Dockerfile: %w", err)
+	}
+	defer file.Close()
+
+	var gitCloneCmd string
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, "RUN git clone") {
+			// Strip the "RUN " prefix
+			gitCloneCmd = strings.TrimSpace(strings.TrimPrefix(line, "RUN "))
+			break
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("failed to read Dockerfile: %w", err)
+	}
+	if gitCloneCmd == "" {
+		return errors.New("no 'RUN git clone' command found in Dockerfile")
+	}
+
+	log.Printf("resolving Swift package for %s/%s:%s", plugin.org, plugin.name, plugin.newVersion)
+
+	// Execute the git clone command
+	cmd := exec.CommandContext(ctx, "sh", "-c", gitCloneCmd)
+	cmd.Dir = versionDir
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to run git clone: %w", err)
+	}
+
+	// Extract the repository name from the git clone command to determine the directory
+	parts := strings.Fields(gitCloneCmd)
+	var repoDir string
+	for _, part := range parts {
+		if strings.HasPrefix(part, "https://") {
```

Comment:
just a heuristic for the repo in the `git clone` line; we can loosen this over time if it doesn't work.